### PR TITLE
Add final 3.0.0 release notes

### PR DESCRIPTION
**Pull request by DomAyre**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,116 @@ All notable changes to this project will be documented in this file.
...
+- The previous logging macros (`LOG_INFO_FMT`, `LOG_DEBUG_FMT` etc) have been deprecated, and should no longer be used by application code. Replace with the `CCF_APP_*` equivalent.
+- Added a new method `get_decoded_request_path_param`s that returns a map of decoded path parameters (#4126).
+- New `crypto::hmac` API (#4204).
+- The `ccf::RpcContext` now contains functionality for storing user data with `set_user_data` and retrieving it with `get_user_data` (#4291).
+- There are now `make_endpoint_with_local_commit_handler` and `make_read_only_endpoint_with_local_commit_handler` functions to install endpoints with post local-commit logic (#4296).
+- `ccf::historical::adapter`, `ccf::historical::adapter_v1`, `ccf::historical::is_tx_committed` and `ccf::historical::is_tx_committed_v1` have been removed. Application code should upgrade to `ccf::historical::adapter_v3` and `ccf::historical::is_tx_committed_v2`.
+- `ccf::EnclaveAttestationProvider` is deprecated and will be removed in a future release. It should be replaced by `ccf::AttestationProvider`.
```
Comment (achamayou) [inline CHANGELOG.md:24]:
```suggestion
- `ccf::EnclaveAttestationProvider` is deprecated and will be removed in a future release. It is replaced by `ccf::AttestationProvider`.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,116 @@ All notable changes to this project will be documented in this file.
...
+
+### Documentation
+
+- The "Node Output" page has been relabelled as "Troubleshooting" in the documentation and CLI commands for troubleshooting have been added to it.
+
+---
+
+### Fixes
+
+- Fix issue with large snapshots that may cause node crash on startup (join/recover) if configured stack size was too low (#4566).
```
Comment (jumaffre) [inline CHANGELOG.md:109]: I think this should be moved under the `Operations` section, and the `Fixed` subheading can be removed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,116 @@ All notable changes to this project will be documented in this file.
...
+- New `crypto::hmac` API (#4204).
+- The `ccf::RpcContext` now contains functionality for storing user data with `set_user_data` and retrieving it with `get_user_data` (#4291).
+- There are now `make_endpoint_with_local_commit_handler` and `make_read_only_endpoint_with_local_commit_handler` functions to install endpoints with post local-commit logic (#4296).
+- `ccf::historical::adapter`, `ccf::historical::adapter_v1`, `ccf::historical::is_tx_committed` and `ccf::historical::is_tx_committed_v1` have been removed. Application code should upgrade to `ccf::historical::adapter_v3` and `ccf::historical::is_tx_committed_v2`.
+- `ccf::EnclaveAttestationProvider` is deprecated and will be removed in a future release. It should be replaced by `ccf::AttestationProvider`.
+- The functions `starts_with`, `ends_with`, `remove_prefix`, and `remove_suffix`, and the type `remove_cvref` have been removed from `nonstd::`. The C++20 equivalents should be used instead.
+- CCF is now a separate CMake project and Debian package per platform (sgx, snp and virtual), rather than the same project and package with a decorated version, to prevent accidental misuse and narrow down dependencies. (#4421).
```
Comment (achamayou) [inline CHANGELOG.md:26]: I think the last three entries should move to the top, they're the most likely to impact users.

**Review by achamayou — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (DomAyre) [reply CHANGELOG.md]: https://github.com/microsoft/CCF/pull/4632/commits/c983985e8236a5cf3b4e28425a7b09f4b9098817

File: CHANGELOG.md (hunk 3 above)
Comment (DomAyre) [reply CHANGELOG.md]: https://github.com/microsoft/CCF/pull/4632/commits/34fb3604b0156cc9bf18138de83e5a65fb682fd3

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,117 @@ All notable changes to this project will be documented in this file.
...
+
+---
+
+#### Release artefacts
+
+- CCF is now a separate CMake project and Debian package per platform (sgx, snp and virtual), rather than the same project and package with a decorated version, to prevent accidental misuse and narrow down dependencies. (#4421).
+  - C++ applications should find the appropriate CCF package in CMake with `find_package("ccf_<platform>" REQUIRED)`.
+  - CCF Debian packages are now installed at `/opt/ccf_<platform>` rather than `/opt/ccf`.
+- `ccf_unsafe` is now a separate project and package, rather than the same project and package with a decorated version, to prevent accidental misuse.
+- Release assets now include variants per TEE platform: `ccf_sgx_<version>_amd64.deb`, `ccf_snp_<version>_amd64.deb` and `ccf_virtual_<version>_amd64.deb`.
```
Comment (achamayou) [inline CHANGELOG.md:22]:
```suggestion
- Release assets now include variants per platform: `ccf_sgx_<version>_amd64.deb`, `ccf_snp_<version>_amd64.deb` and `ccf_virtual_<version>_amd64.deb`.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,117 @@ All notable changes to this project will be documented in this file.
...
+- Docker images now include variants per TEE platform, identified via image tag: `:<version>-sgx`, `:<version>-snp` and `:<version>-virtual`.
```
Comment (achamayou) [inline CHANGELOG.md:23]:
```suggestion
- Docker images now include variants per platform, identified via image tag: `:<version>-sgx`, `:<version>-snp` and `:<version>-virtual`.
```

virtual isn't a TEE

**Review by jumaffre — APPROVED** [review]

**Review by eddyashton — APPROVED** [review]

**Review by MahatiC — APPROVED** [review]

**Pull request merged**